ui/pages/settings/security-tab/delete-metametrics-data-button/delete-metametrics-data-button.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { CONSENSYS_PRIVACY_LINK } from '../../../../../shared/lib/ui-utils';
+import ClearMetametricsData from '../../../../components/app/clear-metametrics-data';
+import {
+  Box,
+  ButtonPrimary,
+  Icon,
+  IconName,
+  IconSize,
+  PolymorphicRef,
+  Text,
+} from '../../../../components/component-library';
+import {
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  getMetaMetricsDataDeletionTimestamp,
+  getMetaMetricsDataDeletionStatus,
+  getMetaMetricsId,
+  getShowDataDeletionErrorModal,
+  getShowDeleteMetaMetricsDataModal,
+  getLatestMetricsEventTimestamp,
+} from '../../../../selectors';
+import { openDeleteMetaMetricsDataModal } from '../../../../ducks/app/app';
+import DataDeletionErrorModal from '../../../../components/app/data-deletion-error-modal';
+import { formatDate } from '../../../../helpers/utils/util';
+import { DeleteRegulationStatus } from '../../../../../shared/constants/metametrics';
+
+const DeleteMetaMetricsDataButton = ({
+  wrapperRef,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Let's find a way to remove the use of `any`. What is the error you see that caused you use `any`?